### PR TITLE
Integrate record loader

### PR DESCRIPTION
**Pull request by JamesGlover**
Migrates record-loader to Gem

### CODE REVIEW

File: spec/data/record_loader/barcode_printer_types/two_entry_example.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
```
Comment (andrewsparkes) [inline spec/data/record_loader/barcode_printer_types/two_entry_example.yml:1]: All your yml files are called 'two_entry_example' which doesn't functionally matter but they should probably be renamed to something more relevant

File: spec/data/record_loader/plate_purposes/two_entry_example.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (andrewsparkes) [inline spec/data/record_loader/plate_purposes/two_entry_example.yml:1]: filename

File: spec/data/record_loader/plate_types/two_entry_example.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+---
```
Comment (andrewsparkes) [inline spec/data/record_loader/plate_types/two_entry_example.yml:1]: filename

File: spec/data/record_loader/tag_group_adapter_types/two_entry_example.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (andrewsparkes) [inline spec/data/record_loader/tag_group_adapter_types/two_entry_example.yml:1]: filename

File: spec/lib/record_loader/barcode_printer_type_loader_spec.rb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+# This file was initially generated via `rails g record_loader`
+RSpec.describe RecordLoader::BarcodePrinterTypeLoader, type: :model, loader: true do
+  subject(:record_loader) do
+    described_class.new(directory: test_directory, files: selected_files)
+  end
+
+  # Tests use a separate directory to avoid coupling your specs to the data
+  let(:test_directory) { Rails.root.join('spec/data/record_loader/barcode_printer_types') }
+
+  context 'with two_entry_example selected' do
```
Comment (andrewsparkes) [inline spec/lib/record_loader/barcode_printer_type_loader_spec.rb:15]: more refs to 'two entry example'

File: spec/lib/record_loader/plate_type_loader_spec.rb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+RSpec.describe RecordLoader::PlateTypeLoader, type: :model, loader: true do
+  subject(:record_loader) do
+    described_class.new(directory: test_directory, files: selected_files)
+  end
+
+  # Tests use a separate directory to avoid coupling your specs to the data
+  let(:test_directory) { Rails.root.join('spec/data/record_loader/plate_types') }
+
+  context 'with two_entry_example selected' do
+    let(:selected_files) { 'two_entry_example' }
```
Comment (andrewsparkes) [inline spec/lib/record_loader/plate_type_loader_spec.rb:16]: and here

File: spec/lib/record_loader/tag_group_adapter_type_loader_spec.rb [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+RSpec.describe RecordLoader::TagGroupAdapterTypeLoader, type: :model, loader: true do
+  subject(:record_loader) do
+    described_class.new(directory: test_directory, files: selected_files)
+  end
+
+  # Tests use a separate directory to avoid coupling your specs to the data
+  let(:test_directory) { Rails.root.join('spec/data/record_loader/tag_group_adapter_types') }
+
+  context 'with two_entry_example selected' do
+    let(:selected_files) { 'two_entry_example' }
```
Comment (andrewsparkes) [inline spec/lib/record_loader/tag_group_adapter_type_loader_spec.rb:16]: and here

**Comment by JamesGlover** [comment]
> Record loader seems pretty cool and easy to understand.
> Just found some refs to the 'two entry example' throughout that come from the default installed examples and have been copy pasted around. Fix those and merge it I think.

Hmm, you might be right. Its not so much copy pasted, but actually generated by the scaffold, which inserts that test. I'll make sure I do a similar change to the gem once I've thought of a better naming pattern.

**Pull request merged**